### ISSUE TITLE
squashing pi in core.py

### ISSUE THREAD

**Issue by neveryoubetray** [openai/spinningup#101]
Hello, 

For my understanding, I wanted to confirm in the sac algorithm, in core.py we apply a squashing function to put mu and pi in a tanh, then in line 77 when we calculate logp_pi between 0 and 1, "to avoid evil machine precision error", we use 1 - pi**2. Should we be using logp_pi here? We calculated the Gaussian log likelihood earlier, but then don't seem to use it (logp_pi is also an argument for apply_squashing_func()).

Thanks for your time!

**Comment by hardikmeisheri** [comment]
`logp_pi -= tf.reduce_sum(tf.log(clip_but_pass_gradient(1 - pi**2, l=0, u=1) + 1e-6), axis=1)`

We are reducing from the earlier calculated value of logp_pi, not assigning it.
